plugin/documents.py

```diff
@@ -529,80 +515,19 @@ def _on_color_boxes(self, response: Any) -> None:
     def on_code_lens_capability_registered_async(self) -> None:
         self._do_code_lenses_async()
 
-    def _code_lens_key(self, index: int) -> str:
-        return self.CODE_LENS_KEY + str(index)
-
-    def _render_code_lens(self, name: str, index: int, region: sublime.Region, command: Optional[Command]) -> None:
-        if command is not None:
-            command_name = command.get("command")
-            if command_name:
-                annotation = make_command_link("lsp_execute", command["title"], {
-                    "session_name": name,
-                    "command_name": command_name,
-                    "command_args": command.get("arguments")
-                })
-            else:
-                annotation = command["title"]
-        else:
-            annotation = "..."
-        annotation = '<div class="codelens">{}</div>'.format(annotation)
-        accent = self.view.style_for_scope("region.greenish markup.accent.codelens.lsp")["foreground"]
-        self.view.add_regions(self._code_lens_key(index), [region], "", "", 0, [annotation], accent)
-
     def _do_code_lenses_async(self) -> None:
         session = self.session_async("codeLensProvider")
         if session and session.uses_plugin():
-            params = {"textDocument": text_document_identifier(self.view)}
             for sv in self.session_views_async():
                 if sv.session == session:
-                    for request_id, request in sv.active_requests.items():
-                        if request.method == "codeAction/resolve":
-                            session.send_notification(Notification("$/cancelRequest", {"id": request_id}))
-            name = session.config.name
-            session.send_request_async(
-                Request("textDocument/codeLens", params, self.view),
-                lambda r: self._on_code_lenses_async(name, r))
-
-    def _on_code_lenses_async(self, name: str, response: Optional[List[CodeLens]]) -> None:
-        for i in range(0, len(self._code_lenses)):
-            self.view.erase_regions(self._code_lens_key(i))
-        self._code_lenses.clear()
-        if not isinstance(response, list):
-            return
-        for index, c in enumerate(response):
-            region = range_to_region(Range.from_lsp(c["range"]), self.view)
-            self._code_lenses.append((c, region))
-            if "command" in c:
-                # We consider a code lens that has a command to be already resolved.
-                self._on_resolved_code_lens_async(name, index, region, c)
-            else:
-                self._render_code_lens(name, index, region, None)
-        self._code_lenses = list((c, range_to_region(Range.from_lsp(c["range"]), self.view)) for c in response)
-        self._resolve_visible_code_lenses_async()
+                    sv.start_code_lenses_async()
```

Comment:
Because the data is now stored in SessionView, can't we now request code lenses for _all_ sessions attached to the view?